sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -479,7 +354,287 @@ private[hive] trait HiveInspectors {
+            poi.getWritableConstantValue.getHiveDecimal)
+      case poi: WritableConstantTimestampObjectInspector =>
+        data: Any => {
+          val t = poi.getWritableConstantValue
+          t.getSeconds * 1000000L + t.getNanos / 1000L
+        }
+      case poi: WritableConstantIntObjectInspector =>
+        data: Any =>
+          poi.getWritableConstantValue.get()
+      case poi: WritableConstantDoubleObjectInspector =>
+        data: Any =>
+          poi.getWritableConstantValue.get()
+      case poi: WritableConstantBooleanObjectInspector =>
+        data: Any =>
+          poi.getWritableConstantValue.get()
+      case poi: WritableConstantLongObjectInspector =>
+        data: Any =>
+          poi.getWritableConstantValue.get()
+      case poi: WritableConstantFloatObjectInspector =>
+        data: Any =>
+          poi.getWritableConstantValue.get()
+      case poi: WritableConstantShortObjectInspector =>
+        data: Any =>
+          poi.getWritableConstantValue.get()
+      case poi: WritableConstantByteObjectInspector =>
+        data: Any =>
+          poi.getWritableConstantValue.get()
+      case poi: WritableConstantBinaryObjectInspector =>
+        data: Any => {
+          val writable = poi.getWritableConstantValue
+          val temp = new Array[Byte](writable.getLength)
+          System.arraycopy(writable.getBytes, 0, temp, 0, temp.length)
+          temp
+        }
+      case poi: WritableConstantDateObjectInspector =>
+        data: Any =>
+          DateTimeUtils.fromJavaDate(poi.getWritableConstantValue.get())
+      case mi: StandardConstantMapObjectInspector =>
+        val keyUnwrapper = unwrapperFor(mi.getMapKeyObjectInspector)
+        val valueUnwrapper = unwrapperFor(mi.getMapValueObjectInspector)
+        data: Any => {
+          // take the value from the map inspector object, rather than the input data
+          val keyValues = mi.getWritableConstantValue.asScala.toSeq
+          val keys = keyValues.map(kv => keyUnwrapper(kv._1)).toArray
+          val values = keyValues.map(kv => valueUnwrapper(kv._2)).toArray
+          ArrayBasedMapData(keys, values)
+        }
+      case li: StandardConstantListObjectInspector =>
+        val unwrapper = unwrapperFor(li.getListElementObjectInspector)
+        data: Any => {
+          // take the value from the list inspector object, rather than the input data
+          val values = li.getWritableConstantValue.asScala
+            .map(unwrapper)
+            .toArray
+          new GenericArrayData(values)
+        }
+      case poi: VoidObjectInspector =>
+        data: Any =>
+          null // always be null for void object inspector
+      case pi: PrimitiveObjectInspector => pi match {
+        // We think HiveVarchar/HiveChar is also a String
+        case hvoi: HiveVarcharObjectInspector if hvoi.preferWritable() =>
+          data: Any => {
+            if (data != null) {
```

Comment:
There are many data: Any => { if (data != null) { ... ... }}. how about use one function to define it to decrease
 codes.